patterns/java/src/refactoring_guru/patterns/cash_prototype/example/Application.java

```diff
@@ -0,0 +1,14 @@
+package refactoring_guru.patterns.cash_prototype.example;
+
+import refactoring_guru.patterns.prototype.example.Rectangle;
+
+public class Application {
+    public static void main(String[] args) {
```

Comment:
Would you please structure App class like this:

constructor()
// .. create cache object, assign it to a field

main()
// request a shape, show it contents

I'm asking this to better illustrate that you create cache object once and then use it everywhere.